File: engine/src/main/java/com/terralite/engine/entity/EntityId.java
package com.terralite.engine.entity;

public record EntityId(long value) {
    public EntityId {
        if (value <= 0) {
            throw new IllegalArgumentException("Entity id must be positive");
        }
    }

    public static EntityId of(long value) {
        return new EntityId(value);
    }
}

File: engine/src/commonMain/kotlin/com/terralite/engine/entity/EntityId.kt
package com.terralite.engine.entity

@JvmRecord
data class EntityId(val value: Long) {
    init {
        require(value > 0) { "Entity id must be positive" }
    }

    companion object {
        @JvmStatic fun of(value: Long): EntityId = EntityId(value)
    }
}
